pkg/common/namespace_selection.go

```diff
@@ -323,17 +304,22 @@ func (r *NamespaceSelectorReconciler) update(namespace string, name string, sel
 }
 
 func filter(allNSList corev1.NamespaceList, t policyv1.Target) ([]string, error) {
-	// If MatchLabels and MatchExpressions are nil, the resulting label selector matches all namespaces. This is to
-	// guard against that.
-	if t.MatchLabels == nil && t.MatchExpressions == nil && len(t.Include) == 0 {
+	// If MatchLabels and MatchExpressions are nil, the resulting label selector
+	// matches all namespaces. This is to guard against that.
+	if t.LabelSelector == nil && len(t.Include) == 0 {
 		return []string{}, nil
 	}
 
-	labelSelector := parseToLabelSelector(t)
+	// List all namespaces by default, otherwise use provided LabelSelector
+	selector := labels.Everything()
```

Comment:
Apparently a `nil` LabelSelector now matches no namespaces, so we have to make it explicit here now.